recipes/llvm/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+from conans import ConanFile, tools, CMake
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
+import os, shutil, glob
+
+projects = [
+    'clang',
+    'clang-tools-extra',
+    'compiler-rt',
+    'debuginfo-tests',
+    'libc',
+    'libclc',
+    'libcxx',
+    'libcxxabi',
+    'libunwind',
+    'lld',
+    'lldb',
+    'mlir',
+    'openmp',
+    'parallel-libs',
+    'polly',
+    'pstl'
+]
+
+default_projects = [
+    'clang',
+    'compiler-rt'
+]
+
+class Llvm(ConanFile):
+    name = 'llvm'
+    description = 'The LLVM Project is a collection of modular and reusable compiler and toolchain technologies'
+    url = 'https://github.com/conan-io/conan-center-index'
+    homepage = 'https://github.com/llvm/llvm-project'
+    license = 'Apache 2.0'
+    topics = 'conan', 'c++', 'compiler', 'tooling'
+
+    settings = 'os', 'arch', 'compiler', 'build_type'
+
+    no_copy_source = True
+    _source_subfolder = 'source_subfolder'
+
+    options = {**{ 'with_' + project : [True, False] for project in projects }, **{
+        'fPIC': [True, False]
+    }}
+    default_options = {**{ 'with_' + project : project in default_projects for project in projects }, **{
+        'fPIC': True
+    }}
+    generators = 'cmake_find_package'
+
+    @property
+    def repo_folder(self):
+        return os.path.join(self.source_folder, self._source_subfolder)
+
+    def project_folder(self, project):
+        return os.path.join(self.repo_folder, project)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = 'llvm-project-llvmorg-' + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, '14')
+
+        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "19.1":
+            raise ConanInvalidConfiguration("Need MSVC >= 19.1")
+
+    def build(self):
+        enabled_projects = [project for project in projects if getattr(self.options, 'with_' + project)]
+        self.output.info('Enabled LLVM subprojects: {}'.format(', '.join(enabled_projects)))
+
+        cmake = CMake(self);
+        cmake.configure(
+            defs = {
+                'LLVM_ENABLE_PROJECTS': ';'.join(enabled_projects)
```

Comment:
Maybe also add `LLVM_ENABLE_BINDINGS: False`. These are probably not needed in a C++ library, and the OCaml bindings end up in /usr/lib/ocaml` by default (at least on my system) which is outside of the conan cache folder.